webapp/src/extensionsBrowser.tsx

```diff
@@ -0,0 +1,453 @@
+        const installed: ExtensionMeta[] = []
+        const reposToFetch: string[] = [];
+        Object.keys(pkg.mainPkg?.deps as Object).forEach((k) => {
+            if (k == "this" || k == "core") {
+                return;
+            }
+            const ext = pkg.mainPkg.deps[k];
+            if (ext?.installedVersion?.includes("github")) {
+                const match = /github:(\S*)#?/.exec(ext.installedVersion);
+                const repoName = match[1]
+
+                let fetchedRepo = getExtensionFromFetched(k);
+
+                if (fetchedRepo) {
+                    installed.push(fetchedRepo)
+                } else {
+                    reposToFetch.push(repoName)
+                }
+            } else {
+                installed.push({
+                    name: ext?.config?.name,
+                    imageUrl: ext?.config?.icon,
+                    description: ext?.config?.description
+                })
+            }
+        })
+
+        if (reposToFetch && reposToFetch.length > 0) {
+            fetchGithubDataAndAddAsync(reposToFetch, (ext) => setInstalledExtensions([...installed, ...ext]))
+        }
+        setInstalledExtensions(installed)
+    }
+
+    function handleImportUrl(url: string) {
+        setShowImportExtensionDialog(false)
+        props.hideExtensions()
+        const ext = getExtensionFromFetched(url)
+        if (!ext) {
+            fetchGithubDataAndAddAsync([url], (exts) => addExtensionsToPool(exts))
+        } else {
+            addGithubPackage(ext)
+        }
+    }
+
+
+    const categoryNames = getCategoryNames();
+    const local = currentTab == TabState.InDevelopment ? fetchLocalRepositories() : undefined
+
+    return <div className={`extensionsBrowser ${props.isVisible ? "" : "hide"}`} >
+        {showImportExtensionDialog ? <ImportModal onCancelClick={() => setShowImportExtensionDialog(false)} onImportClick={handleImportUrl} /> : undefined}
+        {deletionCandidate ? <DeleteConfirmationModal ns={deletionCandidate.name} onCancelClick={() => { setDeletionCandidate(undefined) }} onDeleteClick={() => { removeDepAsync(deletionCandidate)}} /> : undefined}
+        <MenuBar className="extensionsHeader" ariaLabel={lf("Extentions")}>
+            <div className="header-left">
+                <Button className="menu-button" leftIcon="fas fa-arrow-left large" title={lf("Back")} label={lf("Back")} onClick={props.hideExtensions} />
+            </div>
+        </MenuBar>
+        <div className="extensionSearchHeader">
+            <div className="header">{(lf("Do more with your micro:bit"))}</div>
+            <SearchInput searchHandler={setSearchFor} />
+            <div className="extensionTags">
+                {categoryNames.map(c =>
+                    <div className={"extensionTag " + (selectedTag == c ? "selected" : "")} onClick={() => handleCategoryClick(c)}>{c}</div>
+                )}
+            </div>
+            {/* <div className="importButton">
```

Comment:
delete?

Reply:
I want to bring in the import button.....eventually. But the modal was kinda tricky to do in time. Should I still delete and just add back later?

Reply:
you can leave it, just add a comment!